### DISCUSSION TITLE
Persistent and/or Parallel Connections

### DISCUSSION THREAD

**Discussion by sarogersz** [khoih-prog/AsyncHTTPRequest_Generic#24] [Q&A]
I don't see any mention of persistent or parallel connections in the documentation for AsyncHTTPRequest_Generic.  I would like to understand what are the fastest and most efficient methods of use when using the library with http triggering events that overlap in the period between request and response.  Is it necessary to finish a prior request/response cycle before starting another?

**Comment by khoih-prog** [comment]
Thanks for your interest in the library.

> I would like to understand what are the fastest and most efficient methods of use when using the library with http triggering events that overlap in the period between request and response. Is it necessary to finish a prior request/response cycle before starting another?

I'm afraid nobody has spent much time on this or similar libraries to measure, investigate and have answer to your special questions and use-case.

I'd appreciate if you can investigate, then post the results here to help everybody, including me.

Thanks and Regards,

**Comment by sarogersz** [comment]
Thanks for the response.  I am working on using the library and can get  by without making overlapping requests, but I can try it sometime and post what I find out.  Shouldn’t be too hard to try.  I would like to understand this library better.  It could be quite useful in making code efficient when using HTTP.

Stephen

**Comment by sarogersz** [comment]
Looking at the example code and headers, it appears that the library can only handle sequential requests.  Adding this would be a potentially significant step up in complexity, so in keeping with the idea that AsyncHTTPRequest_Generic is a **"simple HTTP layer on top of the ESPAsyncTCP/AsyncTCP/STM32 AsyncTCP library to facilitate REST communication from a Client to a Serve**r" I wouldn't expexct that this is a current or planned future capability.
